PowerNSX.psm1

```diff
@@ -20212,6 +20212,76 @@ function Remove-NsxIpSet {
 }
 
 
+function Remove-NsxIpPool {
+
+    <#
+    .SYNOPSIS
+    Removes the specified NSX IPPool.
+
+    .DESCRIPTION
+    An IP Pool is a simple IPAM construct in NSX that simplifies automated IP
+    address asignment for multiple NSX technologies including VTEP interfaces
+    NSX Controllers.
+
+    This cmdlet removes the specified IP Pool. If the object 
+    is currently in use the api will return an error.
+
+    .EXAMPLE
+    PS C:\> Get-NsxIPPool TestIPPool | Remove-NsxIPPool
+
+    #>
+ 
+    param (
+
+        [Parameter (Mandatory=$true,ValueFromPipeline=$true,Position=1)]
+            [ValidateNotNullOrEmpty()]
+            [System.Xml.XmlElement]$IPPool,
+        [Parameter (Mandatory=$False)]
+            #Prompt for confirmation.  Specify as -confirm:$false to disable confirmation prompt
+            [switch]$Confirm=$true,
+        [Parameter (Mandatory=$False)]
+            #PowerNSX Connection object.
+            [ValidateNotNullOrEmpty()]
+            [PSCustomObject]$Connection=$defaultNSXConnection
+
+
+    )
+    
+    begin {
+
+    }
+
+    process {
+
+        if ($ippool.SelectSingleNode("descendant::extendedAttributes/extendedAttribute[name=`"isReadOnly`" and value=`"true`"]")) {
```

Comment:
Due to differences in the way XML handling must be handled on PowerShell Core, we cant use the native .SelectSingleNode (or SelectNodes) method of the XMLElement.  There is an abstraction function introduced called invoke-xpathquery to allow this to work on either platform.  Need to change this query to utilise invoke-xpathquery.